src/Requires.jl

```diff
@@ -54,7 +54,8 @@ end
 if isprecompiling()
     precompile(loadpkg, (Base.PkgId,)) || @warn "Requires failed to precompile `loadpkg`"
     precompile(withpath, (Any, String)) || @warn "Requires failed to precompile `withpath`"
-    precompile(err, (Any, Module, String)) || @warn "Requires failed to precompile `err`"
+    precompile(err, (Any, Module, String, String, Int)) || @warn "Requires failed to precompile `err`"
+    precompile(err, (Any, Module, String, String, Nothing)) || @warn "Requires failed to precompile `err`"
```

Comment:
We could clear out all the warnings (now that Julia 1.7 does this automatically), but that could be a separate PR.